src/components/Menu/Menu/Menu.tsx

```diff
@@ -87,15 +87,20 @@ const Menu: VibeComponent<MenuProps> & {
 
     const children = useMemo(() => {
       const allChildren = React.Children.toArray(originalChildren);
```

Comment:
I propose we cast the type here for all the array, like this
```suggestion
      const allChildren = React.Children.toArray(originalChildren) as MenuChild[];
```
And let's define MenuChild in `MenuConstants` and reuse it, also let's make the type field optional to avoid the errors, should be smth like this:
```
type MenuChild = ReactElement & {
  type?: Record<string, unknown>;
};
```